src/app/features/tech-record/components/tyres-search/tyres-search.component.ts

```diff
@@ -0,0 +1,171 @@
+  public searchResults: Array<ReferenceDataTyre> | null = null;
+  public vehicleTechRecord?: VehicleTechRecordModel;
+  public viewableTechRecord: TechRecordModel | undefined = undefined;
+  private params: Params = {};
+  private pageStart?: number;
+  private pageEnd?: number;
+  public itemsPerPage: number = 10;
+
+  public template: FormNode = {
+    name: 'criteria',
+    type: FormNodeTypes.GROUP,
+    children: [
+      {
+        name: 'filter',
+        label: 'Search filter',
+        value: '',
+        type: FormNodeTypes.CONTROL
+      },
+      {
+        name: 'term',
+        value: '',
+        type: FormNodeTypes.CONTROL
+      }
+    ]
+  };
+
+  ngOnInit() {
+    this.form = this.dfs.createForm(this.template) as CustomFormGroup;
+    this.globalErrorService.clearErrors();
+    this.route.params.subscribe(p => (this.params = p));
+    this.technicalRecordService.editableTechRecord$.subscribe(data => (this.viewableTechRecord = data));
+    this.referenceDataService
+      .getTyreSearchReturn$()
+      .pipe(take(1))
+      .subscribe((data: any) => {
+        this.searchResults = data;
+      });
+    if (!this.viewableTechRecord) {
+      this.router.navigate(['../..'], { relativeTo: this.route });
+    }
+  }
+
+  get roles() {
+    return Roles;
+  }
+  get currentVrm(): string | undefined {
+    return this.vehicleTechRecord?.vrms.find(vrm => vrm.isPrimary === true)?.vrm;
+  }
+  get paginatedFields(): ReferenceDataTyre[] {
+    return this.searchResults!.slice(this.pageStart, this.pageEnd);
+  }
+  get numberOfResults(): number {
+    return this.searchResults?.length!;
+  }
+
+  handleSearch(term: string, filter: string): void {
+    this.globalErrorService.clearErrors();
+    this.searchResults = [];
+    term = term.trim();
+
+    if (!term) {
+      this.globalErrorService.addError({ error: 'You must provide search criteria', anchorLink: 'term' });
+      return;
+    } else if (!filter) {
+      this.globalErrorService.addError({ error: 'You must select a valid search filter', anchorLink: 'term' });
+      return;
+    }
+
+    if (filter === 'code') {
+      this.referenceDataService.loadReferenceDataByKeySearch(ReferenceDataResourceType.Tyres, term);
+    } else {
+      this.referenceDataService.loadTyreReferenceDataByKeySearch(filter, term);
+    }
+
+    this.actions$
+      .pipe(
+        ofType(fetchReferenceDataByKeySearchSuccess, fetchTyreReferenceDataByKeySearchSuccess),
+        mergeMap(_action => this.store.select(selectTyreSearchReturn())),
+        take(1)
+      )
+      .subscribe((data: any) => {
+        this.router.navigate(['.'], { relativeTo: this.route, queryParams: { 'search-results-page': 1 } });
+        this.searchResults = data;
+      });
+  }
+
+  handleAddTyreToRecord(tyre: ReferenceDataTyre): void {
+    if (this.viewableTechRecord) {
+      const axleIndex = Number(this.params.axleNumber!) - 1;
+      this.viewableTechRecord = cloneDeep(this.viewableTechRecord);
```

Comment:
Doesn't feel very type safe having all these `!` in this function. This could be a way of checking tyres is defined and removing all those `!`
```suggestion
  handleAddTyreToRecord(tyre: ReferenceDataTyre): void {
  const axleIndex = Number(this.params.axleNumber!) - 1;
    if (this.viewableTechRecord && this.viewableTechRecord.axles[axleIndex].tyres) { 
      this.viewableTechRecord = cloneDeep(this.viewableTechRecord);
```

Reply:
The page redirects you if you do not have a viewable tech record, so it has to be safe, as it cannot even be on this page without existing. Unsure about the params but assume the same as they are passed in with the page, they must be there

Reply:
Sorry I meant the ones afetr the tyres not the viewable tech record